### PR TITLE
rpcserver: Add script version to gettxout.

### LINKED ISSUE #2647: rpcserver: Add `Version` field to `gettxout.scriptPubKey` results.
As the title indicates, a `version` field should be added to the result of the `gettxout` RPC under `scriptPubKey`.  The script version is going to become more important in the future when a new version is introduced.

### PR DESCRIPTION
**Pull request by JoeGruffins**
closes #2647

### CODE REVIEW

**Comment by JoeGruffins** [comment]
So, the version was moved from `tylpes.GetTxOutResult` to `types.ScriptPubKeyResult`?

**Review by davecgh — CHANGES_REQUESTED** [review]
Please update the [JSON-RPC API](https://github.com/decred/dcrd/blob/master/docs/json_rpc_api.mediawiki) docs as well.

**Comment by davecgh** [comment]
> So, the version was moved from `tylpes.GetTxOutResult` to `types.ScriptPubKeyResult`?

Not really.  They're different.  The version that existed on `types.GetTxOutResult` was the **transaction** version, whereas this the **script** version.

**Comment by JoeGruffins** [comment]
gettxoutreturns

**Review by davecgh — APPROVED** [review]

File: docs/json_rpc_api.mediawiki [hunk 1]
```diff
@@ -2031,6 +2031,13 @@ of the best block.
...
+:: <code>asm</code>: <code>(string)</code> Disassembly of the script.
+:: <code>hex</code>: <code>(string)</code> Hex-encoded bytes of the script. Omitted if empty.
+:: <code>reqSigs</code>: <code>(numeric)</code> The number of required signatures. Omitted if empty.
+:: <code>type</code>: <code>(string)</code> The type of the script (e.g. 'pubkeyhash').
+:: <code>addresses</code>: <code>(json array of strings)</code> The Decred addresses associated with this script. Omitted if empty.
+:: <code>commitamt</code>: <code>(numeric)</code> The ticket commitment value if the script is for a staking commitment. Omitted if empty.
+:: <code>version</code>: <code>(string)</code> The script version.
 : <code>coinbase</code>: <code>(numeric)</code> Whether or not the transaction is a coinbase.
 |-
 !Example Return
```
Comment (rstaudt2) [inline docs/json_rpc_api.mediawiki:2043]: This example return could be updated to include the script version as well.

**Review by rstaudt2 — APPROVED** [review]

File: docs/json_rpc_api.mediawiki (hunk 1 above)
Comment (JoeGruffins) [reply docs/json_rpc_api.mediawiki:2043]: ah

Comment (JoeGruffins) [reply docs/json_rpc_api.mediawiki:2043]: Should it be mainnet? I suppose I could just add `,"version": 0`

Comment (davecgh) [reply docs/json_rpc_api.mediawiki:2043]: I'd prefer it to be mainnet so it matches the rest of the examples in the doc.  All that really needs to change, imo, from the original example is to add `,"version": 0` as you noted.

**Review by davecgh — APPROVED** [review]

**Pull request merged**